web-app/js/portal/ui/openlayers/control/SpatialConstraint.js

```diff
@@ -65,11 +67,131 @@ Portal.ui.openlayers.control.SpatialConstraint = Ext.extend(OpenLayers.Control.D
         });
     },
 
-    _onSketchStarted: function(e) {
+    _isRegularPolygon: function() {
+        return (this.handler.line === undefined || this.handler.line == null);
+    },
 
-        this.triggerMapClick(e);
-        if (this.map.mapPanel) {
-            this.map.mapPanel._closeFeatureInfoPopup();
+    _polygonCoordsReset: function() {
+        this.polygonCoords = []
+    },
+
+    _addPolygonCoord: function(lonLat) {
+        this.polygonCoords.push(lonLat);
+    },
+
+    _polygonLastCoord: function() {
+        return this.polygonCoords[this.polygonCoords.length-1]
+    },
+
+    _polygonCoordsExist: function() {
+        return this.polygonCoords.length >= 1;
+    },
+
+    _antimeridianPointCross: function(lon1, lon2) {
+        if ((lon1 > 0 && lon2 > 0) || (lon1 < 0 && lon2 < 0)) {
+            return false;
+        }
+        return true;
+    },
+
+    _polygonClickInSameLocation: function(lonLat1, lonLat2) {
+        if (lonLat1.lat == lonLat2.lat && lonLat1.lon == lonLat2.lon) {
+            return true;
+        }
+        return false;
+    },
+
+    _polygonRemoveDuplicatePointsGeom: function(geometry) {
+        var lastxy = {x: -1, y: -1};
+        var removeIndex = [];
+
+        geometry.components[0].components.forEach(function(component, i) {
+            var thisxy = {x: component.x, y: component.y}
+            if (thisxy.x == lastxy.x && thisxy.y == lastxy.y) {
+                removeIndex.push(i);
+            }
+            lastxy = thisxy;
+        });
+
+        removeIndex.reverse();
+        removeIndex.forEach(function(i) {
+            geometry.components[0].components.splice(i,1);
+        });
+
+        return geometry;
+    },
+
+    _shiftMapCentre: function(lon) {
+        var center = this.map.getCenter();
+
+        if ((center.lon > 0 && lon < 0) || (center.lon < 0 && lon > 0)) {
```

Comment:
This function is doing two separate things here - checking if there is a cross over the antimeridian for the centre and a supplied lon value, and then returning a bool if the centre has moved. Maybe pull out the comparo into a separate method, and then keep this function as a straight shift. Better from a testing point of view